draft-ietf-quic-qpack.md

```diff
@@ -251,14 +299,33 @@ Because QUIC does not guarantee order between data on different streams, a
 header block might reference an entry in the dynamic table that has not yet been
 received.
 
-Each header block contains a Largest Reference which identifies the table state
-necessary for decoding. If the greatest absolute index in the dynamic table is
-less than the value of the Largest Reference, the stream is considered
-"blocked."  While blocked, header field data should remain in the blocked
-stream's flow control window.  When the Largest Reference is zero, the frame
-contains no references to the dynamic table and can always be processed
-immediately. A stream becomes unblocked when the greatest absolute index in the
-dynamic table becomes greater than or equal to the Largest Reference for all
+Each header block contains a Largest Reference field which identifies the
+table state necessary for decoding.
+
+When the Largest Reference value is not specified, the frame contains no
+references to the dynamic table and can always be processed immediately.
+
+When it is specified, the Largest Reference value is compared with
+the absolute index value of the next entry to be inserted into the
+dynamic table.  That entry's absolute index value divides the absolute
```

Comment:
This explanation, relying on the "next entry to be inserted" is awkward.  As I suggested, if you count the total number of inserts, you get something easier to manage:

```
0 -----
  First: Insert (i.e., Index 8)
1 -----
...
  Index: 1
8 -----
  Index: 0
9 -----  << Base offset = 2 (or -2, I forget which this is).
  Post-Base: 0
10 -----
  Post-Base: 1
11 -----  << Table state = 11
  This: Can't be referenced
12 -----
```

Then you can say that the signaled table state and the decoder's view of the table state might differ, but only by less than `MaxEntries`.  That means that if `SignaledTableState%(MaxEntries*2)` differs from `LocalTableState` by more than MaxEntries, then it needs to be wrapped to get the full table state.

Don't worry about describing these in terms of absolute values.  I think that's easier than having the variables be modulo `2MaxEntries`.

Reply:
I used the "next entry to be inserted" because I see no other way to describe the initial state, when there is no "last inserted entry."  Perhaps we could add the concept of "cursor?"

The "total number of inserts" concept eludes me.  The diagram is confusing, as it combines both encoder instructions (`Insert`) and header block instructions (`Post-Base`).

1. If the "total number of inserts" is the number of inserts performed while encoding a given header block, how does the decoder know where to start?
1. If this number is added to the current absolute index, then it is the same as _Largest Reference_.
1. ?